### PR TITLE
Transfered Shop and Dropzone Loading to be handled by Map.

### PR DESCRIPTION
**Pull request by J-Lin1996**
Moved creation of shops and drop zone to map.cpp
Randomized shop positioning
Edited map asset
NOTE: 0s used when loading drop zones will be replaced when drop zone functionality completed.

### CODE REVIEW

**Review by JohnAgapeyev — APPROVED** [review]
We have talked about the SRAND call
So otherwise it is all good

File: src/game/GameManager.h [hunk 1]
```diff
@@ -118,6 +119,12 @@ class GameManager {
     void freeDropPoint(const int32_t id);
     bool checkFreeDropPoints();
 
+    // Ai Map setters and getters
+    auto& getAiMap() const { return AiMap; };
+    void setAiMap(const std::array<std::array<bool, M_WIDTH>, M_HEIGHT> a) {
```
Comment (JohnAgapeyev) [inline src/game/GameManager.h]: Needs to be passed by reference

File: src/map/Map.cpp [hunk 2]
```diff
@@ -161,12 +162,31 @@ void Map::genWalls(const vector<MapPoint>& wallStart) {
...
+    // Shop position being loaded.
+    MapPoint shopPosition;
+
     for (const auto& w : walls) {
-        logv("Creating wall");
         GameManager::instance()->createWall(w.x, w.y, w.width, w.height);
     }
+
+    // Random number generation.
+    ran.seed(1);
```
Comment (JohnAgapeyev) [inline src/map/Map.cpp]: There is a default seed used in the constructor
No need to manually seed it unless you want to guarantee that the shop and drop zones will be identical across every run by using the same seed

**Review by JohnAgapeyev — APPROVED** [review]
Read comments

File: src/map/Map.h [hunk 3]
```diff
@@ -57,12 +58,11 @@ class Map {
     void mapLoadToGame();
     void printData();
 
-    void setFile(const string f) { fname = f; };
+    void setFile(const std::string f) { fname = f; };
```
Comment (isaacmorneau) [inline src/map/Map.h]: const reference

**Review by isaacmorneau — CHANGES_REQUESTED** [review]
only one thing

File: src/map/Map.cpp (hunk 2 above)
Comment (J-Lin1996) [reply src/map/Map.cpp]: I removed the seed, but it still gives me the same number each time it runs... not too sure why, but we can fix it later.

File: src/map/Map.h (hunk 3 above)
Comment (J-Lin1996) [reply src/map/Map.h]: Changed to reference

**Review by isaacmorneau — APPROVED** [review]

**Pull request merged**